src/metpy/io/metar.py

```diff
@@ -176,8 +182,7 @@ def parse_metar_to_dataframe(metar_text, year=datetime.now().year, month=datetim
     return df
 
 
-def parse_metar_to_named_tuple(metar_text, station_metadata, year=datetime.now().year,
-                               month=datetime.now().month):
+def parse_metar_to_named_tuple(metar_text, station_metadata, year, month):
```

Comment:
Curious what is happening with this change?  `year` and `month` now both appear to be required and not included in the function parameter docs?

Reply:
`parse_metar_to_named_tuple` is not public and is currently only used within the two modified public functions, and so always inherits some year and month from them based on the described behavior in this PR. In `_to_named_tuple`, year and month are not optional to construct `date_time` and return the named tuple. This is why we have this behavior of a hard default to the current year and month on `None`. I can update the docstring for `_to_named_tuple` for our own knowledge for sure, and there are plenty of ways we could open this up to a larger discussion on parsing datetime, as in #1327.

Reply:
Thanks for keeping an eye on things @akrherz !